app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/dialogs/SelectionActionWidget.java

```diff
@@ -189,8 +215,15 @@ private View createSeparator() {
     }
 
     private void handleAction(View sender) {
+        String action = (String)sender.getTag();
+        if (action.equals(ACTION_WEB_SEARCH) && mSelectionText != null) {
+            mWidgetManager.getWindows().addTab(
+                    mWidgetManager.getWindows().getFocusedWindow(),
+                    UrlUtils.urlForText(getContext(), mSelectionText));
+        }
+
```

Comment:
Can we modify `addTab()` so that if the current window has a Session, it becomes the parent of the new Session so that the back button works. I want to be able to do a search with google, the hit the back button and go back to what I was looking at. I think we should do this with `what's new` and any other button that creates a new tab with web content but that can be a follow up.

Reply:
Good call. I've added that.